### PR TITLE
Fixes #19413 win.navigator may be null | undefined

### LINKED ISSUE #19413: FluentUI Not work in SSR mode, win.navigator.userAgent is not safe
### Environment Information

- **Package version(s)**: (fill this out)
    8.28.1
- **Browser and OS versions**: (fill this out if relevant)

### Please provide a reproduction of the bug in a codepen:
I want to use FluentUI in a kind of ssr framework. But when i run the code, it throws a error 
image
I deep in this, and I find i need to setSSR to render fluentUI in ssr mode. But when i setSSR(true), it didnot work.
image
image
image
image

I find in source code, there is a check in isMac function, but it didnot check win.navigator is undefined wheather or not. Now most ssr framework mock window in server side, so it throwed that error.

I have two problems:
 (1) setSSR is not work
 (2) win.navigator.userAgent is not safe。 

### Priorities and help requested:

Are you willing to submit a PR to fix? (Yes, No)
Yes
Requested priority: (Blocking, High, Normal, Low)
High

Products/sites affected: (if applicable)

### PR DESCRIPTION
**Pull request by XYjourney**
#### Pull request checklist

- [x] Addresses an existing issue: Fixes #19413
- [ ] Include a change request file using `$ yarn change`

#### Description of changes

(give an overview)

#### Focus areas to test

(optional)

### CODE REVIEW

File: packages/utilities/src/osDetector.ts
Comment (XYjourney) [reply packages/utilities/src/osDetector.ts]: Thanks, have updated~

**Comment by ecraig12345** [comment]
@XYjourney You'll need to sign the CLA as linked in this comment before we can merge the PR. https://github.com/microsoft/fluentui/pull/19542#issuecomment-908057204

**Review by ecraig12345 — APPROVED** [review]
Approved once the CLA is signed

**Comment by XYjourney** [comment]
> @XYjourney You'll need to sign the CLA as linked in this comment before we can merge the PR. [#19542 (comment)](https://github.com/microsoft/fluentui/pull/19542#issuecomment-908057204)
I am a Microsoft employee. I have been a member in MS org in github.

**Pull request merged**
